File: data/translated_projects/commons-fileupload/src/main/org/apache/commons/fileupload/RequestContext.py
from __future__ import annotations
from abc import ABC
from io import StringIO
import io
from io import BytesIO
import typing
from typing import *
import java # type: ignore
from src.main.org.apache.commons.fileupload.java_handler import *
class RequestContext(ABC, metaclass=StaticFieldRedirector):
    def getInputStream(self) -> typing.Union[io.BytesIO, io.StringIO, io.BufferedReader]:
        pass


    def getContentLength(self) -> int:
        pass


    def getContentType(self) -> str:
        pass


    def getCharacterEncoding(self) -> str:
        pass


    javaClz = java.type("org.apache.commons.fileupload.RequestContext")
    def __getattr__(self, name):
        try:
            return object.__getattribute__(self, name)
        except AttributeError:
            pass
        return JavaHandler.mapping(getattr(self.javaClz, StaticFieldRedirector.unmangle_name(name)))

File: logs/glue/commons-fileupload/src/main/java/org/apache/commons/fileupload/RequestContext.java
package org.apache.commons.fileupload;

import java.io.IOException;
import java.io.InputStream;
import org.apache.commons.fileupload.ContextInitializer;
import org.apache.commons.fileupload.ExceptionHandler;
import org.apache.commons.fileupload.IntegrationUtils;
import org.graalvm.polyglot.PolyglotException;
import org.graalvm.polyglot.Value;

public interface RequestContext {
  InputStream getInputStream() throws IOException;

  int getContentLength();

  String getContentType();

  String getCharacterEncoding();
}
